src/components/playback/playbackmanager.js

```diff
@@ -1789,42 +1790,90 @@ class PlaybackManager {
             });
         }
 
-        function translateItemsForPlayback(items, options) {
-            if (!items.length) return Promise.resolve([]);
+        async function translateItemsForPlayback(items, options) {
+            if (!items.length) return [];
 
-            if (items.length > 1 && options && options.ids) {
+            sortItemsIfNeeded(items, options);
+
+            const firstItem = items[0];
+            const serverId = firstItem.ServerId;
+            const queryOptions = options.queryOptions || {};
+
+            const promise = getPlaybackPromise(firstItem, serverId, options, queryOptions, items);
+
+            if (promise) {
+                const result = await promise;
+                return result ? result.Items : items;
+            } else {
+                return items;
+            }
+        }
+
+        function sortItemsIfNeeded(items, options) {
+            if (items.length > 1 && options?.ids) {
                 // Use the original request id array for sorting the result in the proper order
                 items.sort(function (a, b) {
                     return options.ids.indexOf(a.Id) - options.ids.indexOf(b.Id);
                 });
             }
+        }
 
-            const firstItem = items[0];
-            let promise;
-
-            const serverId = firstItem.ServerId;
+        function getPlaybackPromise(firstItem, serverId, options, queryOptions, items) {
+            const nonItemTypePromise = getNonItemTypePromise(firstItem, serverId, options, queryOptions);
```

Comment:
I was afraid the splitting would cause problems.
Currently, when you start a series/season, it falls into [`IsFolder`](https://github.com/jellyfin/jellyfin-web/blob/588da3ce95a851a3ada5ba812d15c36d039dbc24/src/components/playback/playbackmanager.js#L1909-L1925) branch (series/season is a folder, but with a corresponding type) and doesn't reach [`Series/Season`](https://github.com/jellyfin/jellyfin-web/blob/588da3ce95a851a3ada5ba812d15c36d039dbc24/src/components/playback/playbackmanager.js#L1865-L1868).

Moving the `getNonItemTypePromise` call after `switch` would fix this, but I'm not sure it wouldn't cause another problem.

Reply:
What if I added additional checks on the `isFolder` if statement that also checks it doesn't have type `Series` or `Season`

Reply:
> What if I added additional checks on the `isFolder` if statement that also checks it doesn't have type `Series` or `Season`

While this would solve the problem, it would make the logic more complex and the code less maintainable. _I'm not saying the current code of web doesn't look like spaghetti._ 😅

I think we should consider the "IsFolder" if statement as a fallback. In this case, moving the "non-type" promise getter after the "typed" block makes more sense.

In fact, we need to test from stricter cases to more general cases.

Reply:
Alright I swapped the order and it seems like everything works. Tried testing with movies/season/series playing an individual episode along with playing a folder. Worked with photo slideshows as well. Just double check I didn't miss anything.